source/installation_guide/dovecot_community_repositories/compiling_source.rst

```diff
@@ -0,0 +1,220 @@
+Patches to fix it are welcome.
+
+OpenSSL is used by default now, and it should be automatically detected.
+If it is not, you are missing some header files or libraries, or they
+are just in a non-standard path. Make sure you have the ``openssl-dev``
+or a similar package installed, and if it is not in the standard
+location, set ``CPPFLAGS`` and ``LDFLAGS`` as shown in the first
+section above:
+
+.. code-block:: none
+
+   CPPFLAGS="-I/opt/openssl/include" LDFLAGS="-L/opt/openssl/lib" ./configure
+
+By default the SSL certificate is read from
+``/etc/ssl/certs/dovecot.pem`` and the private key from
+``/etc/ssl/private/dovecot.pem``. The ``/etc/ssl`` directory can be
+changed using the ``--with-ssldir=DIR`` configure option. Both can of
+course be overridden from the configuration file.
+
+Solaris and OpenSSL problems
+****************************
+
+Solaris 10 includes a bundled OpenSSL that does not function correctly
+with Dovecot when attempting to use SSL/TLS with the default dovecot
+config. This is because the default setting of ssl_cipher_list in
+``dovecot.conf`` is HIGH:!ALL; due to import restrictions in some
+countries (now apparently relaxed) the high level routines are part of
+the unbundled SUNWcry package and are not available if you don't have
+this package installed. This confuses the client as dovecot announces
+support for high level crypto and then cannot deliver. In any case, to
+resolve this you can alternatively (in decreasing order of simplicity):
+
+1. Set ``ssl_cipher_list = MEDIUM:!LOW`` in ``dovecot.conf``
+
+2. Find and install the missing SUNWcry package.
+
+3. Provide an alternate version of the openssl libraries that doesn't
+   have the high grade routines removed for your protection (sigh). The
+   bundled version of OpenSSL cannot be removed. Installing a newer
+   OpenSSL from source or package (for instance, from
+   `<http://sunfreeware.com/>`_) will enable Dovecot to work correctly
+   as long as you link against the new OpenSSL. Assuming you are
+   building with the built-in ld, make and gcc, then your build should
+   go something like this (notice the -R required by Sun's linker that
+   sets the runtime linking path in the resulting programs so the
+   OpenSSL libraries load from ``/usr/local/ssl/lib``):
+
+.. code-block:: none
+
+   PATH=$PATH:/usr/sfw/bin:/usr/ccs/bin
+   export PATH
+   mv /usr/lib/pkgconfig/openssl.pc /usr/lib/pkgconfig/openssl.pc.orig
+   CPPFLAGS=-I/usr/local/ssl/include \
+    LDFLAGS='-L/usr/local/ssl/lib -R/usr/local/ssl/lib' \
+    ./configure --with-ssl=openssl
+   make
+   make install
+
+Notify method
+^^^^^^^^^^^^^
+
+Linux
+*****
+
+Note that current ``inotify`` is in the Linux kernel since version
+2.6.13 and it is preferred over ``dnotify``. If your distribution does
+not have the required ``inotify`` header file, you can get it from the
+inotify maintainer (this example requires `cURL <http://curl.haxx.se/>`_):
```

Comment:
This whole notify method stuff contains old old obsolete stuff. I think we can just remove it all.